### PR TITLE
Add recorder state to console

### PR DESCRIPTION
**Pull request by pljones**
#351 Improve recorder logging

The big bits of re-doing the changes to the recorder are still awaiting me having time to give to them, so I've just put this in server.cpp for now.  I'll sort it out with the rest.

Does this need anything in Changelog?

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -705,7 +707,12 @@ void CServer::SetEnableRecording ( bool bNewEnableRecording )
 {
     if ( bRecorderInitialised )
     {
+        // note that this block executes regardless of whether
+        // what appears to be a change is being applied, to ensure
+        // the requested state is the result
+
         bEnableRecording = bNewEnableRecording;
+        qInfo() << "Recording state" << ( bEnableRecording ? "enabled" : "disabled" );
```
Comment (corrados) [inline src/server.cpp:715]: You need to check the Qt version to use qInfo: https://doc.qt.io/qt-5/qtglobal.html#qInfo

**Comment by corrados** [comment]
No need for ChangeLog entry.

File: src/server.cpp (hunk 1 above)
Comment (corrados) [reply src/server.cpp:715]: See, e.g., here: https://github.com/corrados/jamulus/blob/master/src/clientdlg.cpp#L763

File: src/server.cpp [hunk 2]
```diff
@@ -405,7 +406,7 @@ CServer::CServer ( const int          iNewMaxNumChan,
     if ( !strRecordingDirName.isEmpty() )
     {
         bRecorderInitialised = JamRecorder.Init ( this, iServerFrameSizeSamples );
-        bEnableRecording     = bRecorderInitialised;
+        SetEnableRecording ( bRecorderInitialised );
```
Comment (corrados) [inline src/server.cpp:409]: Why call the function? If bRecorderInitialised is false, the function body in SetEnableRecording is not entered. If it is true, the special implementation in the if-condition is not reached. So there is no difference in just applying the new value to the boolean variable. Do you have something else in mind why you changed this? Do you want to put new functionality in the SetEnableRecording function?

File: src/server.cpp (hunk 1 above)
Comment (pljones) [reply src/server.cpp:715]: Ah right, will do.

File: src/server.cpp (hunk 2 above)
Comment (pljones) [reply src/server.cpp:409]: Ah...  Ahh.....  AAHH!!! Yes... very good point.  Oops.

Comment (pljones) [reply src/server.cpp:409]: ...except if it's true, I want the message, and I'd rather have the code for the message only once...

Comment (corrados) [reply src/server.cpp:409]: Ok, makes sense. Thanks for the explaination.

**Pull request merged**